pg_bm25/src/api/search.rs

```diff
@@ -120,8 +121,15 @@ pub fn minmax_bm25(
 
 #[pg_extern]
 fn drop_bm25_internal(index_name: &str) {
-    // Drop the Tantivy data directory.
     let writer_client = WriterGlobal::client();
+    if needs_commit() {
+        writer_client
+            .lock()
+            .expect("could not lock writer on drop_bm25")
+            .request(crate::writer::WriterRequest::Commit)
+            .expect("error committing existing transaction during drop_bm25");
+    }
+    // Drop the Tantivy data directory.
```

Comment:
This is unrelated to the fix, but realized it was necessary as I wrote the test.